chia/wallet/wallet_node.py

```diff
@@ -310,7 +310,8 @@ def set_server(self, server: ChiaServer):
             self.config["peer_connect_interval"],
             self.log,
         )
-        asyncio.create_task(self.wallet_peers.start())
+        if "full_node_peer" not in self.config:
```

Comment:
We might have full node peer in config, but we are not actually running a full node. In that case we still need to start the node discovery.